feat: add `lapack/base/dlartg` (#7596)
stdlib-js/stdlib: lib/node_modules/@stdlib/lapack/base/dlartg/README.md

<!--

@license Apache-2.0

Copyright (c) 2025 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# dlartg

> > LAPACK routine to construct a Givens rotation of a two-dimensional vector that zeros out the second component.

Reviewer: @aayush0325 I'm a little confused. What is meant by a "two-dimensional" vector? Vectors are, by definition, one dimensional. Are you meaning a column matrix?

Reviewer: Or maybe you are intending a two-element vector?

Author: i mean a column matrix here
<img width="811" height="207" alt="image" src="https://github.com/user-attachments/assets/4ed9f93c-6a7d-454c-93bb-3850a51dd60b" />
[ F ]
[ G ]

this is what im referring to as a 2 dimensional vector, i was a little confused here as well so i decided to go with this.

Author: let me know if i should use something else here please i don't really know what else to use

<section class="intro">

`dlartg` constructs a Givens rotation for a two-dimensional vector such that the following equation is satisfied:

> 

Reviewer: ```suggestion
This routine constructs a Givens rotation for a two-dimensional vector such that the following equation is satisfied:
```

```javascript
var dlartg = require( '@stdlib/lapack/base/dlartg' );
```

#### dlartg( F, G, C, S, R )

Constructs a Givens rotation of a two-dimensional vector that zeros out the second component. For a two-dimensional vector `( F, G )` it computes the sine `S`, cosine `C` and the resultant `R`.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var F = new Float64Array( [ 3.0 ] );
var G = new Float64Array( [ 4.0 ] );
var C = new Float64Array( 1 );
var S = new Float64Array( 1 );
var R = new Float64Array( 1 );

dlartg( F, G, C, S, R );
// R => <Float64Array>[ 5.0 ]
// C => <Float64Array>[ 0.6 ]
// S => <Float64Array>[ 0.8 ]
```

The function has the following parameters:

-   **F**: single element array representing the first component of the vector to be rotated as a [`Float64Array`][mdn-float64array].
-   **G**: single element array representing the second component of the vector to be rotated as a [`Float64Array`][mdn-float64array].
-   **C**: single element array overwritten by the cosine of the rotation as a [`Float64Array`][mdn-float64array].
-   **S**: single element array overwritten by the sine of the rotation as a [`Float64Array`][mdn-float64array].

> -   **R**: single element array overwritten by the length of the rotated vector as a [`Float64Array`][mdn-float64array].

Reviewer: ```suggestion
-   **F**: [`Float64Array`][mdn-float64array] containing a single element representing the first component of the vector to be rotated.
-   **G**: [`Float64Array`][mdn-float64array] containing a single element representing the second component of the vector to be rotated
-   **C**: [`Float64Array`][mdn-float64array] containing a single element which is overwritten by the cosine of the rotation.
-   **S**: [`Float64Array`][mdn-float64array] containing a single element which is overwritten by the sine of the rotation.
-   **R**: [`Float64Array`][mdn-float64array] containing a single element which is overwritten by the length of the rotated vector.
```

var C1 = new Float64Array( C0.buffer, C0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
var S1 = new Float64Array( S0.buffer, S0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
var R1 = new Float64Array( R0.buffer, R0.BYTES_PER_ELEMENT*1 ); // start at 2nd element

dlartg( F1, G1, C1, S1, R1 );
// R0 => <Float64Array>[ 0.0, 5.0 ]
// C0 => <Float64Array>[ 0.0, 0.6 ]
// S0 => <Float64Array>[ 0.0, 0.8 ]
```

#### dlartg.ndarray( F, offsetF, G, offsetG, C, offsetC, S, offsetS, R, offsetR )

Constructs a Givens rotation of a two-dimensional vector that zeros out the second component, using alternative indexing semantics. For a two-dimensional vector `( F, G )` it computes the sine `S`, cosine `C` and the resultant `R`.

```javascript
var Float64Array = require( '@stdlib/array/float64' );

var F = new Float64Array( [ 3.0 ] );
var G = new Float64Array( [ 4.0 ] );
var C = new Float64Array( 1 );
var S = new Float64Array( 1 );
var R = new Float64Array( 1 );

dlartg.ndarray( F, 0, G, 0, C, 0, S, 0, R, 0 );
// R => <Float64Array>[ 5.0 ]
// C => <Float64Array>[ 0.6 ]
// S => <Float64Array>[ 0.8 ]
```

The function has the following parameters:

> -   **F**: single element array representing the first component of the vector to be rotated as a [`Float64Array`][mdn-float64array].

Reviewer: Same comments as above.

> -   **offsetF**: starting index for `F`.

Reviewer: ```suggestion
-   **offsetF**: index of the element in `F`.
```

It is somewhat awkward to say "starting index" when we are talking about only a single element. We can be more precise, IMO.

Applies here and below and likely elsewhere throughout this PR.
